Geometry/ForwardCommonData/plugins/dd4hep/DDBHMAngular.cc

```diff
@@ -0,0 +1,71 @@
+#include "DataFormats/Math/interface/GeantUnits.h"
+#include "DetectorDescription/Core/interface/DDSplit.h"
+#include "DetectorDescription/DDCMS/interface/DDPlugins.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DD4hep/DetFactoryHelper.h"
+
+//#define EDM_ML_DEBUG
+using namespace geant_units::operators;
+
+static long algorithm(dd4hep::Detector& /* description */,
+                      cms::DDParsingContext& ctxt,
+                      xml_h e,
+                      dd4hep::SensitiveDetector& /* sens */) {
+  cms::DDNamespace ns(ctxt, e, true);
+  cms::DDAlgoArguments args(ctxt, e);
+  int units = args.value<int>("number");
+  double rr = args.value<double>("radius");
+  double dphi = args.value<double>("deltaPhi");
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("ForwardGeom") << "DDBHMAngular debug: Parameters for positioning-- " << units
+                                  << " copies at radius " << convertMmToCm(rr) << " cm with delta(phi) "
+                                  << convertRadToDeg(dphi);
+#endif
+  std::string rotMat = args.value<std::string>("Rotation");
+  std::string childName = args.value<std::string>("ChildName");
+  dd4hep::Volume parent = ns.volume(args.parentName());
+  dd4hep::Volume child = ns.volume(ns.prepend(childName));
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("ForwardGeom") << "DDBHMAngular debug: Parent " << parent.name() << "\tChild " << child.name()
+                                  << "\tRotation matrix " << rotMat;
+#endif
+  std::string rotstr = DDSplit(rotMat).first;
+  dd4hep::Rotation3D rot;
+  if (rotstr != "NULL") {
+    rot = ns.rotation(rotMat);
+  }
+  static const double fac1 = 0.5;
+  static const double fac2 = 1.5;
+  static const double fac3 = 14.5;
+  static const double fac4 = 15.5;
+  for (int jj = 0; jj < units; jj++) {
+    double driverX(0), driverY(0), driverZ(0);
+    if (jj < 16) {
```

Comment:
It would be nice to replace the 16 with a constant. I'm not sure what the 16 means. Then line 49 would be something like `else if (jj == startPos + 1)` if `startPos` was the name of the constant. Then lines 52 and 55 would be similar with +2 and +3.